### DISCUSSION TITLE
Reorganize menus

### DISCUSSION THREAD

**Discussion by Jean-Gabriel1982** [GrandOrgue/grandorgue#686] [General]
Hello everyone,

I am currently on MacOs and the v0.3.1.2341-11.os version of GrandOrgue.

In order to better navigate the program, I allow myself to propose a reorganization of the menus.
Indeed, currently it is a bit confusing, with redundant elements in several menus.

For greater clarity, wouldn't it be logical to reorganize these menus, and also be able to have fewer sections, and therefore fewer clicks?

I suggest that in the "FILE" menu:

- Delete EITHER the "Load" menu, OR the "Open" menu. I think both do double duty.
In fact, when you open the program for the first time, you click on "Install organ package" and in the future, you will click on "Load", unless you want to install a new organ package.
For my part, it seems that "Open" is useless.

- "Settings" should be placed just above the "Import settings" menu. So the "Settings" will be together. It makes more sense to me.

- "Install organ Package" should be placed at the top of the "File" menu.
So we would have: "Install organ package", then "Load", then "Favorite", then "Open recent"

- "Organ properties" would go in a new main menu specific to the organ in use, menu which could for example, take the name of the organ.
For example: Burea Church, or Cavaillé-Coll Caen, etc ...

In this menu specific to the current organ, we would therefore have:
- "Organ Settings" from the "Audio / Midi" menu
- "Temperament" from the "Audio / Midi" menu
- The whole "Panel" menu, which contains configuration items for the organ in use
- "Panic" from the "Audio / Midi" menu
- "Memory set" from the "Audio / Midi" menu
- "Organ properties" from the "File" menu

The rest of the "Audio / Midi" menu items would go in the "Settings" of the "File" menu, which already contain Audio and Midi items.

Finally, in "Settings" of the "File" menu, couldn't we merge the "Options", "Defaults and Initial settings" and "Reverb" tabs? In my opinion this would make sense and limit the tabs.

Likewise, merge the "Organs" and "Organ packages" sections ...

Sorry for the length of the message. Maybe you will find it not very coherent, and bad ... far from me the idea of ​​giving lessons, I just modestly try to make my contribution to this wonderful building that is GrandOrgue :)

**Comment by oleg68** [comment]
Thank you for raising this topic.

I agree that the current menu is quite incomfortable, but not all your propositions are evident.

Let's to separate two issues: reorganising the menu and reorganising the settings dialog. Please open a separate discussion about the dialog and welcom to continue about the menu.

I think we discuss and draw the target menu structure in this topic together and then I'll implement these changes.

First of all, GO can work with two formats of organs
1. Unpacked: a directory containing an ODF file (`.organ`) as well as othe files referenced by the ODF: sound samples, images and so on.
2. Packed - a single `.orgue` file that is an archive containing all necessary components.

GO can use unpacked organs without previous installations. The menu entry `Open..` is for loading this kind of organs.

Otherwise, packed organs should be installed prior the using of it. So the menu entries `Install` and `Load` work with them.

Maybe these entries should be renamed, or to have the single entry `Open` that displays a dialog box with tree choices
- `.organ`
- `.orgue`
- `installed`

with selecting a directory or a file name of a already known organ from the list

**Comment by Jean-Gabriel1982** [comment]
Thank you for your answer

> Let's to separate two issues: reorganising the menu and reorganising the settings dialog. Please open a separate discussion about the dialog and welcom to continue about the menu.

Doesn't the fact of creating a separate topic for the same subject risk leading us astray?
Because all this deals with the reorganization of the menus ...

> First of all, GO can work with two formats of organs...

Ok I understand very well.
But then in this case we should just have "Install organ package" which would install both a compressed and an unzipped package.

And "Open", which would open a previously installed organ.
"Load" meanwhile, would be deleted.

We could then suddenly also delete "Open recent", the "Open" menu being able to list the organs recently opened.
In this case, the user would choose which organ to open.

available to discuss other changes proposed in my first message :)

**Comment by oleg68** [comment]
> Doesn't the fact of creating a separate topic for the same subject risk leading us astray? Because all this deals with the reorganization of the menus .

Settings dialog is not a menu.

They are totally different tasks. I can'not do them in parallel. At the first, the menu structure will be changed and at the second - the dialog.  So It's better to discuss them separately.

> "Install organ package" which would install both a compressed and an unzipped package.

Now uncompressed organ doesn't require instalation at all. I'm not sure that enforcing a user to install uncompressed files would be a good idea.

**Reply by Jean-Gabriel1982** [reply]
> 
> Settings dialog is not a menu.
>
> They are totally different tasks. I can'not do them in parallel. At the first, the menu structure will be changed and at the second - the dialog. So It's better to discuss them separately.

Ok I'll create new topic for Dialog box :)

> 
> Now uncompressed organ doesn't require instalation at all. I'm not sure that enforcing a user to install uncompressed files would be a good idea.

Yes of course, it is not a question of making it all more confusing, but really to simplify the life of the users, while being coherent.

**Comment by oleg68** [comment]
I propose the following menu structure:

- File
    - Settings
    - -------
    - Install Organ
    - Load Organ
    - Reload Current Organ
    - -------
    - Import Organ Settings
    - Save Organ Settings
    - Export Settings/Combinations
    - --------
    - Import Combinations
    - --------
    - Load MiDI
    - ---------
    - Exit
- Organ
    - Show Properties
    - Organ Settings
    - Reset to Defaults
    - Midi Objects
    - -----------
    - Memory Set
- Audio/MIDI
    - Temperament
    - Show Output State
    - Log MIDI Events
    - --------
    - Panic
- Panel
    The same as now
- Help
    The same as now

The main changes:
- The entry ``Install Organ`` can open both `.organ` and `.orgue` files. After installing an organ it opens it. If the file has already been installed previously then this entry only opens it
- The operations with cache are excluded from the menu and moved to the Organs Tab of the Settings

How do you find this new structure?

**Comment by Jean-Gabriel1982** [comment]
It's a very interesting structure.

Simply, I would put "Load MIDI" in "Load Midi file" and below "Reload current Organ"
"Import organ combinations" also just below "Save organ settings"
I would put "Settings" just above "Exit"

What do you think about that ?

**Comment by oleg68** [comment]
@Jean-Gabriel1982 

I changed. And previously I forgot the `Preset` entry, - added it to the `Organ` group.

And I cann't implement `Update Cache` in the Organ tab of Settings: this operation requires the organ is loaded. So I put it in the `Organ` group.

- File
    - Install Organ
    - Load Organ
    - Reload Current Organ
    - --------
    - Load MIDI file
    - -------
    - Import Organ Settings
    - Save Organ Settings
    - --------
    - Import Combinations
    - Export Settings/Combinations
    - ---------
    - Settings...
    - Exit
- Organ
    - Show Properties
    - Organ Settings
    - Reset to Defaults
    - Midi Objects
    - -----------
    - Preset
    - Memory Set
    - -----------
    - Update Cache
- Audio/MIDI
    - Temperament
    - Show Output State
    - Log MIDI Events
    - --------
    - Panic
- Panel
    The same as now
- Help
    The same as now

**Comment by Jean-Gabriel1982** [comment]
I think it is much clearer now :)

**Comment by oleg68** [comment]
I think we should discuss this more.

There are two parts of this suggestions:
1. Union loading .organ and .orgue files into the same menu entry
2. Reorganising the menu structure (without mixing Open with Load)

**Comment by thbeckpr** [comment]
I had a discussion earlier this week on organforum.com in which a user said that GO was so he preferred HW. I asked what was buggy and he replied that Load organ failed. When he tried Open organ it worked and he ascribed this to a bug. 

I explained the differences between the formats and that a help file is available. Of course no complex piece of software is completely intuitive. Is there a way to make it clear in the menu that Load is for .orgue format and Open is for .organ format?

Is it really necessary to have an Install function? Does an installed sample set open more quickly than one that is loaded from .orgue format?

**Comment by oleg68** [comment]
> Is it really necessary to have an Install function? Does an installed sample set open more quickly than one that is loaded from .orgue format?

For now, yes. But I'd leave only one menu entry: open. This action should install automatically if this organ is opened the first time.

**Comment by oleg68** [comment]
But from user point of view Open requires selecting a file from the filesystem, but load - selecting from a short list. The second option is more comfortable in most use cases.

The difference between .organ and .orgue is evident for programmers, but it is not evident for ordinar users. Should we enforce the users to understand technical details?

**Reply by thbeckpr** [reply]
No, but their lack of knowledge using the current menu structure is causing them difficulties.
